Adds enhancedDate apis

gamboagirl opened the pull request:
enhancedDate apis with tests commented out

gamboagirl committed:
Adds enhancedDate apis

enhancedDate apis with tests commented out

jamdon commented:
I think you haven't pushed your enhancedDate.js changes to your repository yet. Make sure you've added and committed your changes in your local repository, then pushed/synced them with your github repository. Then, initiate the pull request again.

gamboagirl commented:
Hmmm...I committed them and synced. I think this happened to me last time. I'm out of town til Sundat evening but I'll see if I can jump onto someone's laptop. 

Michelle M. Gamboa

> On Apr 17, 2015, at 13:40, jamdon notifications@github.com wrote:
> 
> I think you haven't pushed your enhancedDate.js changes to your repository yet. Make sure you've added and committed your changes in your local repository, then pushed/synced them with your github repository. Then, initiate the pull request again.
> 
> —
> Reply to this email directly or view it on GitHub.

gamboagirl committed:
Adds enhancedDate apis

The actual apis. Apparently I submitted the test html which I didn't
plan to submit.

gamboagirl commented:
I wasn't able to access the code on my home machine this weekend and had no cell service where I was (Cannon Beach, OR). I just got home and will submit the .js file.

jamdon commented on app/js/enhancedDate.js:
```diff
@@ -1,5 +1,159 @@
+            if (dateItem === undefined) {
+                myDate = new Date();
+            } else {
+                myDate = new Date(dateItem);
```
You only want to pass dateItem to your Date constructor if dateItem is an instance of Date or a millisecond value (typeof dateItem === 'number').

jamdon commented on app/js/enhancedDate.js:
```diff
@@ -1,5 +1,159 @@
+            if (dateItem) {
+                return myDate;
+            } else {
+                return Date.UTC(myDate);
```
myDate is a Date object, so when you pass it to Date.UTC it won't return what you want - Date.UTC wants a year and a month and, optionally, additional, more granular segments of time. It will return milliseconds, which is what you want your function to return here, but a simpler way to accomplish this would be to call `myDate.getTime( );`

jamdon commented on app/js/enhancedDate.js:
```diff
@@ -1,5 +1,159 @@
+            var day = myDate.getDay();
+            switch (day) {
+                case 0:
+                    return "Sunday";
```
This will work - consider using an array with the day name strings instead an using the value returned by getDay( ) to return that value.

jamdon commented on app/js/enhancedDate.js:
```diff
@@ -1,5 +1,159 @@
+            var month = myDate.getMonth();
+            switch (month) {
+                case 0:
+                    return "January";
```
This works - consider instead using an array of month names and using the value returned by getMonth( ) to index into the array and return that value.

jamdon commented on app/js/enhancedDate.js:
```diff
@@ -1,5 +1,159 @@
+        isToday: function() {
+            console.log("isToday");
+            var now = Date.now();
+            if (this.getDate(myDate) == this.getDate(now)) {
```
When you call this.getDate( ) with a truthy value (both myDate and now will evaluate to truthy in this case), it will simply return myDate as a Date object. This means you're comparing if myDate == myDate. What you want to compare is if myDate's year, month, and date are the same as now's year, month, and date. To get "now", just do `var now = new Date( );`

jamdon commented:
:+1:
